vision/objectdetection/detector.go

```diff
@@ -37,27 +37,53 @@ func Build(prep Preprocessor, det Detector, post Postprocessor) (Detector, error
 // Detection returns a bounding box around the object and a confidence score of the detection.
 type Detection interface {
 	BoundingBox() *image.Rectangle
+	NormalizedBoundingBox() []float64
 	Score() float64
 	Label() string
 }
 
 // NewDetection creates a simple 2D detection.
-func NewDetection(boundingBox image.Rectangle, score float64, label string) Detection {
-	return &detection2D{boundingBox, score, label}
+func NewDetection(imageBounds, boundingBox image.Rectangle, score float64, label string) Detection {
+	normBbox := NewNormalizedBoundingBox(imageBounds, boundingBox)
+	return &detection2D{boundingBox, normBbox, score, label}
+}
+
+// NewDetectionWithoutImgBounds creates a simple 2D detection.
+func NewDetectionWithoutImgBounds(boundingBox image.Rectangle, score float64, label string) Detection {
+	return &detection2D{boundingBox, nil, score, label}
+}
+
+// NewNormalizedBoundingBox creates a normalized bounding box from the image bounds and the bounding box.
+func NewNormalizedBoundingBox(imageBounds, boundingBox image.Rectangle) []float64 {
+	if imageBounds.Max.X == 0 || imageBounds.Max.Y == 0 {
+		return nil
+	}
+	return []float64{
+		float64(boundingBox.Min.X) / float64(imageBounds.Max.X),
+		float64(boundingBox.Min.Y) / float64(imageBounds.Max.Y),
+		float64(boundingBox.Max.X) / float64(imageBounds.Max.X),
+		float64(boundingBox.Max.Y) / float64(imageBounds.Max.Y),
+	}
 }
 
 // detection2D is a simple struct for storing 2D detections.
 type detection2D struct {
-	boundingBox image.Rectangle
-	score       float64
-	label       string
+	boundingBox           image.Rectangle
+	normalizedBoundingBox []float64
```

Comment:
Discussion question: there is a dedicated struct, `image.Rectangle` for bounding boxes. Should we create a similar struct for normalized boxes? I can argue either way.

Reply:
SGTM. Thanks for the comment!